android/keyple-plugin/android-nfc/src/main/kotlin/org/eclipse/keyple/plugin/android/nfc/AbstractAndroidNfcReader.kt

```diff
@@ -288,13 +310,34 @@ internal abstract class AbstractAndroidNfcReader : AbstractObservableLocalAutono
         return readerProtocolName == null || protocolsMap.containsKey(readerProtocolName) && protocolsMap[readerProtocolName] == tagProxy?.tech
     }
 
+    override fun onStartDetection() {
+        Timber.d("onStartDetection")
+        if (nfcAdapter == null) {
+            nfcAdapter = NfcAdapter.getDefaultAdapter(contextWeakRef.get()!!)
+        }
+
+        val flags = flags
+
+        val options = options
+
+        Timber.i("Enabling Read Write Mode with flags : $flags and options : $options")
+
+        // Reader mode for NFC reader allows to listen to NFC events without the Intent mechanism.
+        // It is active only when the activity thus the fragment is active.
+        nfcAdapter?.enableReaderMode(contextWeakRef.get(), this, flags, options)
+    }
+
+    override fun onStopDetection() {
+        Timber.d("onStopDetection")
+        nfcAdapter?.disableReaderMode(contextWeakRef.get())
```

Comment:
You should check non nullity on nfcAdapter and context. If the check fails throw an illegal state exception.